sklearn/semi_supervised/_label_propagation.py

```diff
@@ -105,6 +107,16 @@ class BaseLabelPropagation(ClassifierMixin, BaseEstimator, metaclass=ABCMeta):
          for more details.
     """
 
+    _parameter_constraints = {
+        "kernel": [StrOptions({"knn", "rbf"}), callable],
+        "gamma": [Interval(Real, 0, None, closed="left")],
+        "n_neighbors": [Interval(Integral, 0, None, closed="neither")],
+        "alpha": [None, Interval(Real, 0, 1, closed="neither")],
+        "max_iter": [Interval(Integral, 0, None, closed="neither")],
+        "tol": [Interval(Real, 0, None, closed="left")],
+        "n_jobs": [None, Interval(Integral, -1, None, closed="left")],
```

Comment:
Indeed we can use any negative number with `joblib` (it means all CPU cores apart of the number).

```suggestion
        "n_jobs": [None, Integral],
```